magenta-lib/src/main/scala/magenta/tasks/AWS.scala

```diff
@@ -18,15 +18,21 @@ import magenta.{App, DeployReporter, DeploymentPackage, KeyRing, NamedStack, Sta
 
 import scala.collection.JavaConversions._
 
+sealed case class Region(val host: String) {
+  override def toString = host
+}
+
+object DefaultRegion extends Region("eu-west-1") 
+
 trait S3 extends AWS {
   def s3client(keyRing: KeyRing, config: ClientConfiguration = clientConfiguration) =
     new AmazonS3Client(provider(keyRing), config)
 }
 
 trait Lambda extends AWS {
-  def lambdaClient(implicit keyRing: KeyRing) = {
+  def lambdaClient(region: Region)(implicit keyRing: KeyRing) = {
     val client = new AWSLambdaClient(provider(keyRing), clientConfiguration)
-    client.setEndpoint("lambda.eu-west-1.amazonaws.com")
+    client.setEndpoint(s"lambda.${region}.amazonaws.com")
```

Comment:
I'd rather we switch to using the `client.setRegion` notation rather than encoding endpoints in Riff-Raff when the SDK now knows it. I think you can do `.withRegion` on the client.